### ISSUE TITLE
Something broke in an npm dependency

### ISSUE THREAD

**Issue by christianbundy** [ssbc/ssb-db#261]
Using the ssb-server shrinkwrap hides the problem, but unfortunately it looks like something changed in a dependency of ssb-db and now `npm install` is installing something that's breaking the progress notification bits of Patchwork so it thinks it's never done "Loading...".

Really wish we had a lockfile so that we could track this down, spent about 5 hours on it today but I'm not any closer to resolving it -- I've narrowed the problem down to "a dependency of ssb-db" I think but I'm not sure which. Whatever is causing the problem isn't being tested for because `npm test` seems to pass and using ssb-server with this version of ssb-db doesn't cause any breaks.

TL;DR: Patchwork apparently depends on a dependency of ssb-db that had a backward-incompatible change and we don't have a lockfile so I can't `git bisect`. :confused:

**Comment by christianbundy** [comment]
After a few minutes of running Patchwork I get a core dump, but none of them have been helpful. I've seen `nearley` mentioned often but also a few others in the final stack trace.

```
<--- Last few GCs --->

[25990:0x392a319f000]    98295 ms: Scavenge 1945.5 (2083.0) -> 1944.9 (2083.5) MB, 12.2 / 0.0 ms  (average mu = 0.117, current mu = 0.017) allocation failure 
[25990:0x392a319f000]   102532 ms: Mark-sweep 1945.8 (2083.5) -> 1945.1 (2083.5) MB, 4223.9 / 0.1 ms  (average mu = 0.066, current mu = 0.009) allocation failure scavenge might not succeed


<--- JS stacktrace --->

==== JS stack trace =========================================

    0: ExitFrame [pc: 0x562bd8101d0e]
    1: StubFrame [pc: 0x562bd8077256]
Security context: 0x38fc88dc7411 <String[7]: file://>
    2: decode [0x39ac341f3859] [/home/christianbundy/src/ssbc/ssb-db/node_modules/charwise/codec/object.js:~46] [pc=0x32ddf2e34e26](this=0x32178fdc7389 <Object map = 0x3888c2873f89>,0x363d6b9f1a61 <String[97]: KFE512M1.49407166052600000000000000000000"J@jbYzgqUkv74IHQnE64Zg+jI17qSEhYuLGLIZh5p2U4E=.ed25519!>)
  ...
```

```
<--- Last few GCs --->

[27482:0x38096aafc000]   144305 ms: Scavenge 1935.2 (2081.8) -> 1934.4 (2082.3) MB, 14.1 / 0.0 ms  (average mu = 0.146, current mu = 0.081) allocation failure 
[27482:0x38096aafc000]   144328 ms: Scavenge 1935.4 (2082.3) -> 1934.6 (2082.8) MB, 15.0 / 0.0 ms  (average mu = 0.146, current mu = 0.081) allocation failure 
[27482:0x38096aafc000]   144354 ms: Scavenge 1935.6 (2082.8) -> 1934.8 (2083.3) MB, 14.0 / 0.0 ms  (average mu = 0.146, current mu = 0.081) allocation failure 


<--- JS stacktrace --->

==== JS stack trace =========================================

    0: ExitFrame [pc: 0x563610c9ad0e]
Security context: 0x1bde12b47411 <String[7]: file://>
    1: push [0x1bde12b07a09](this=0x3c475c214b49 <JSArray[0]>,0x3c475c214981 <JSArray[1]>)
    2: /* anonymous */ [0x182dbeb62d69] [/home/christianbundy/src/ssbc/patchwork/node_modules/nearley/lib/nearley.js:~265] [pc=0x3ed41f35148f](this=0x3c475c2023f1 <Parser map = 0x2f11becd3f01>,0x3c475c202309 <String[82]: net:fe80::185d:3dfd:23ad:c3d:8008~shs:...

```

**Comment by dominictarr** [comment]
nearly is used in multiserver-address

what do you mean by "using ssb-server shrinkwrap hides the problem"?

**Comment by christianbundy** [comment]
When I depend on ssb-server with a shrinkwrap I don't have this problem, but when I depend on a [branch without shrinkwrap](https://github.com/ssbc/ssb-server/tree/no-shrinkwrap) then `npm install` returns a dependency tree with the bug (which seems to be an ssb-db dependency).

The next time we update the ssb-server shrinkwrap to include new dependencies it looks like it will break Patchwork because of a dependency change in ssb-db. I was using the `no-shrinkwrap` branch because I wanted to update some deep dependencies (aligned-block-file, flumelog-offset, flumedb, etc) but shrinkwrap doesn't allow that (and it's incompatible with `npm link`).
